plugin/Gui/AdminPageApiSettings.php

```diff
@@ -80,9 +80,13 @@ private function addFormModelAPI()
 		$pInputModelApiKey = new InputModelOption('onoffice-settings', 'apikey', $labelKey, 'string');
 		$optionNameKey = $pInputModelApiKey->getIdentifier();
 		$apiKey = get_option($optionNameKey);
-		if (defined('ONOFFICE_CREDENTIALS_ENC_KEY')) {
+		if (defined('ONOFFICE_CREDENTIALS_ENC_KEY')
+			&& defined('ONOFFICE_CREDENTIALS_ENC_KEY') != get_option('onoffice-credentials-enc-key')) {
```

Comment:
Why are you storing the encryption key in the database?

Reply:
@jmaas-onoffice , cause when delete defind const ‘ONOFFICE_CREDENTIALS_ENC_KEY’ in wp_confid.php, in the UI, the token  displayed encrypted , like link video : https://files.fm/f/q4vww4gn3, my solution solved that problem. Do I have permission to save the encryption key in the database?

Reply:
The encryption key only adds security when it is _not_ stored in the database. It only protects the API credentials when someone has only access to the database, because they would then need additional access to the constant or file access for the `wp-config.php`. If we store the encryption key in the database, they only need the database and then we don't need encryption.

It is actually wanted in the requirements to show the raw (encrypted) token when you delete the encryption key. That way the user can see that the token does not match the real token and they will enter the correct one and save it, which should fix the problem.

I noticed that when you removed the ONOFFICE_CREDENTIALS_ENC_KEY, the GUI still said "(encrypted)". Without ONOFFICE_CREDENTIALS_ENC_KEY, the API credentials should _not_ be encrypted.

So the requirements are:
ONOFFICE_CREDENTIALS_ENC_KEY present => Display "(encrypted)" and encrypt the API credentials.
ONOFFICE_CREDENTIALS_ENC_KEY missing => Do not display "(encrypted)" and do not decrypt the API credentials.

Is this clearer? :)